code/modules/client/preference_setup/vore/07_traits.dm

```diff
@@ -25,6 +25,19 @@
 	var/dirty_synth = 0		//Are you a synth
 	var/gross_meatbag = 0		//Where'd I leave my Voight-Kampff test kit?
 
+/datum/preferences/proc/get_custom_bases_for_species(var/new_species)
+	if (!new_species)
+		new_species = species
+	var/list/choices
+	var/datum/species/spec = GLOB.all_species[new_species]
+	if (spec.selects_bodytype == SELECTS_BODYTYPE_SHAPESHIFTER)
+		choices = spec.get_valid_shapeshifter_forms()
+	else if (spec.selects_bodytype == SELECTS_BODYTYPE_CUSTOM)
+		choices = GLOB.custom_species_bases
```

Comment:
these direct choices = without copying are dangerous as youre modifying the list down the line. Since lists are objects, it will modify wherever you got it from originally too, if you dont make it a copy somewhere along the line.

Reply:
ah
noted